language/stdlib/modules/LibraTransactionPublishingOption.move

```diff
@@ -0,0 +1,92 @@
+address 0x1 {
+
+module LibraTransactionPublishingOption {
+    use 0x1::Vector;
+    use 0x1::LibraConfig;
+    use 0x1::LibraTimestamp;
+
+    const SCRIPT_HASH_LENGTH: u64 = 32;
+
+    const ENOT_GENESIS: u64 = 0;
+    const EINVALID_SCRIPT_HASH: u64 = 1;
+
+    /// Defines and holds the publishing policies for the VM. There are three possible configurations:
+    /// 1. No module publishing, only allowlisted scripts are allowed.
+    /// 2. No module publishing, custom scripts are allowed.
+    /// 3. Both module publishing and custom scripts are allowed.
+    /// We represent these as the following resource.
+    struct LibraTransactionPublishingOption {
+        // Only script hashes in the following list can be executed by the network. If the vector is empty, no
+        // limitation would be enforced.
+        script_allow_list: vector<vector<u8>>,
+        // Transactions with sender in this list won't abide to the allow list rules above.
+        sender_allow_list: vector<address>,
+        // Anyone can publish new module if this flag is set to true.
+        module_publishing_allowed: bool,
+    }
+
+    public fun initialize(
+        lr_account: &signer,
+        script_allow_list: vector<vector<u8>>,
+        sender_allow_list: vector<address>,
+        module_publishing_allowed: bool,
+    ) {
+        assert(LibraTimestamp::is_genesis(), ENOT_GENESIS);
```

Comment:
Please change to `LibraTimestamp::assert_genesis()` now that the error refactor has landed. 

Possibly add checks on the LR account to make sure it's the right account as well (`Roles::assert_libra_root(lr_account)`). But I think this is enforced in `publish_new_config` so not totally necessary.